pkg/apis/autoscaling/v1alpha1/pa_lifecycle.go

```diff
@@ -91,7 +92,11 @@ func (pa *PodAutoscaler) ScaleBounds(asConfig *autoscalerconfig.Config) (int32,
 
 	max := asConfig.MaxScale
 	if paMax, ok := pa.annotationInt32(autoscaling.MaxScaleAnnotationKey); ok {
-		max = paMax
+		if maxLimit := asConfig.MaxScaleLimit; maxLimit != 0 {
+			max = int32(math.Min(float64(paMax), float64(maxLimit)))
```

Comment:
Rather than dealing with all these casts:
```suggestion
		if maxLimit := asConfig.MaxScaleLimit; maxLimit != 0 && maxLimit < paMax {
			max = maxLimit
```